legion/engine/physics/physics_statics.cpp

```diff
@@ -341,5 +287,728 @@ namespace legion::physics
+        const math::vec3& transformedB2, const math::vec3& planeANormal, const math::vec3& planeBNormal)
+    {
+        //------------------------ Check if normals created by arcA seperate normals of B --------------------------------------//
+        
+        float planeADotB1 = math::dot(planeANormal, transformedB1); //CBA
+        float planeADotB2 = math::dot(planeANormal, transformedB2); //DBA
+
+        float dotMultiplyResultA = planeADotB1 * planeADotB2;
+
+        if (dotMultiplyResultA > 0.0f )
+        {
+            return false;
+        }
+
+        //------------------------ Check if normals created by arcB seperate normals of A --------------------------------------//
+
+        float planeBDotA1 = math::dot(planeBNormal, transformedA1); //ADC
+        float planeBDotA2 = math::dot(planeBNormal, transformedA2); //BDC
+
+        float dotMultiplyResultB = planeBDotA1 * planeBDotA2;
+
+        if (dotMultiplyResultB > 0.0f )
+        {
+            return false;
+        }
+
+        //------------------------ Check if arcA and arcB are in the same hemisphere --------------------------------------------//
+
+        math::vec3 abNormal = math::cross(transformedA2, transformedB2);
+
+        float planeABDotA1 = math::dot(abNormal, transformedA1);
+        float planeABDotB1 = math::dot(abNormal, transformedB1);
+
+        float dotMultiplyResultAB = planeABDotA1 * planeABDotB1;
+
+        if (dotMultiplyResultAB < 0.0f || math::epsilonEqual(dotMultiplyResultAB,0.0f,math::epsilon<float>()))
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    bool PhysicsStatics::buildInitialHull(const std::vector<math::vec3>& vertices,
+        std::array<math::vec3,6>& supportVertices, std::vector<HalfEdgeFace*>& faces )
+    {
+        //Summary:
+        //[1] Find the 2 most distant vertices in 'support Vertices'
+        //[2] Find the vertex most distant from the line created by the 2 most distance vertices
+        //[3] Create first collider face using that line and the vertex most distant to it
+        //[4] Find the most distant vertex from the plane where the first collider face lies
+        //[5] Create a set of faces connecting the first collider face to the most distant vertex,
+        //this create a tetrahedron shaped collider
+
+        //[1] Find the 2 most distant vertices in 'vertices'
+
+        float mostDistant = std::numeric_limits<float>::lowest();
+
+        math::vec3& firstDistant = supportVertices.at(0);
+        math::vec3& secondDistant = supportVertices.at(1);
+
+        //Iterate through support vertices to find the combination of vertices that represent the 2 most distant points
+        for (int i = 0; i < supportVertices.size(); i++)
+        {
+            math::vec3& first = supportVertices.at(i);
+
+            for (int j = i+1; j < supportVertices.size()-1; j++)
+            {
+                math::vec3& second = supportVertices.at(j);
```

Comment:
same as before, signed unsigned compare, and `at(size_t)` will cause a cast to `size_type`